packages/material-ui/src/Switch/Switch.test.js

```diff
@@ -87,4 +88,50 @@ describe('<Switch />', () => {
 
     expect(getByRole('checkbox')).to.have.property('checked', false);
   });
+
+  describe('with FormControl', () => {
+    describe('enabled', () => {
+      it('should not have the disabled class', () => {
+        const { getByTestId } = render(
+          <FormControl>
+            <Switch data-testid="root" />
+          </FormControl>,
+        );
+
+        expect(getByTestId('root')).not.to.have.class(classes.disabled);
+      });
+
+      it('should be overridden by props', () => {
+        const { getByTestId } = render(
+          <FormControl>
+            <Switch data-testid="root" disabled />
+          </FormControl>,
+        );
+
+        expect(getByTestId('root')).to.have.class(classes.disabled);
+      });
+    });
+
+    describe('disabled', () => {
+      it('should have the disabled class', () => {
+        const { getByTestId } = render(
+          <FormControl disabled>
+            <Switch data-testid="root" />
+          </FormControl>,
+        );
+
+        expect(getByTestId('root')).to.have.class(classes.disabled);
+      });
+
+      it('should be overridden by props', () => {
```

Comment:
Thanks for adding this as well. It's something we can hopefully get rid of or at least warn about.